modules/core/shared/src/main/scala-3/io/circe/derivation/ConfiguredDecoder.scala

```diff
@@ -0,0 +1,161 @@
+package io.circe.derivation
+
+import scala.deriving.Mirror
+import scala.compiletime.constValue
+import Predef.genericArrayOps
+import cats.data.{NonEmptyList, Validated}
+import io.circe.{Decoder, DecodingFailure, ACursor, HCursor}
+
+trait ConfiguredDecoder[A](using conf: Configuration) extends Decoder[A], DerivedInstance[A]:
```

Comment:
Why does the trait have public methods `elemDecoders` and `elemDefaults`?

Reply:
I saw no reason to hide these methods.